lib/latex.js

```diff
@@ -21,14 +24,26 @@ function defineDefaultProperty (target, property) {
   })
 }
 
-export default class Latex {
-  process = new ProcessManager()
-  status = new StatusIndicator()
-  opener = new OpenerRegistry()
+export default class Latex extends Disposable {
+  disposables = new CompositeDisposable()
 
   constructor () {
+    super(() => this.disposables.dispose())
     this.createLogProxy()
     defineDefaultProperty(this, 'logger')
+
+    this.defineImmutableProperty('builderRegistry', new BuilderRegistry())
+    this.defineImmutableProperty('composer', new Composer())
+    this.defineImmutableProperty('opener', new OpenerRegistry())
+    this.defineImmutableProperty('process', new ProcessManager())
+    this.defineImmutableProperty('status', new StatusIndicator())
+  }
+
+  defineImmutableProperty (name, value) {
```

Comment:
Do we need to export this? Can this be moved outside of the class just like `defineDefaultProperty`?